Make slideshow mobile friendly

adampaulukanis opened the pull request:
What do you think about this?

I know it is a mess. Especially style.css file but it is work in progress ;)

### .gitignore 

I noticed that my OS adds some strange files. The 2 last actually comes from VIM.

### index nav hrefs

I thought at first that previous links were OK but I noticed I have a problem with them, so changed to #

Commit:
Make slideshow mobile friendly

What do you think about this?

saqi commented:
Improvement from last commit, well done! :+1: 

The link for Home redirects to the parent directory - if you can add the # for that link as well :)

Very happy about the responsiveness, an improvement I can suggest for that is to use proper images ;)
Also for the carousel to go to next image after a certain number of seconds

I think it would be good if we work from bottom to top, like let's put the logo on and finish the menu (removing underlines and centering menu items)

Further down we can use the original images, remove scrolling and use jQuery to slide up the text; please see link below for complete version :)
https://codepen.io/Saqalain/full/MBLPdm

Commit:
Logo and menu: Add logo and change colours

I do not know what font was used for logo.
Perhaps it would be nice to have a favicon?

adampaulukanis commented:
I found a bug in my code. I do not want to fix it yet, actually I want to,
but I guess I need to dive into my code and probably rewrite it.
When I introduced 'Automatic Slideshow' (change slides every X seconds) I
did not take into account changing slides by clicking prev/next buttons.
Now it does not work.
Next works sort of. I need to clear the timeout and start it again. I will
think about it but now I want to focus on essentials.

On Sun, 7 Oct 2018 at 21:32, Saqlain Quassimali <notifications@github.com>
wrote:

> Improvement from last commit, well done! 👍
>
> The link for Home redirects to the parent directory - if you can add the #
> for that link as well :)
>
> Very happy about the responsiveness, an improvement I can suggest for that
> is to use proper images ;)
> Also for the carousel to go to next image after a certain number of seconds
>
> I think it would be good if we work from bottom to top, like let's put the
> logo on and finish the menu (removing underlines and centering menu items)
>
> Further down we can use the original images, remove scrolling and use
> jQuery to slide up the text; please see link below for complete version :)
> https://codepen.io/Saqalain/full/MBLPdm
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/saqi/Maybank/pull/3#issuecomment-427684609>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABieU8GeBPZZX4EElS3qsmxmINi78hlZks5uimTIgaJpZM4XMA3l>
> .
>

saqi commented:
> I found a bug in my code. I do not want to fix it yet, actually I want to, but I guess I need to dive into my code and probably rewrite it. When I introduced 'Automatic Slideshow' (change slides every X seconds) I did not take into account changing slides by clicking prev/next buttons. Now it does not work. Next works sort of. I need to clear the timeout and start it again. I will think about it but now I want to focus on essentials.
> […](#)
> On Sun, 7 Oct 2018 at 21:32, Saqlain Quassimali ***@***.***> wrote: Improvement from last commit, well done!  The link for Home redirects to the parent directory - if you can add the # for that link as well :) Very happy about the responsiveness, an improvement I can suggest for that is to use proper images ;) Also for the carousel to go to next image after a certain number of seconds I think it would be good if we work from bottom to top, like let's put the logo on and finish the menu (removing underlines and centering menu items) Further down we can use the original images, remove scrolling and use jQuery to slide up the text; please see link below for complete version :) https://codepen.io/Saqalain/full/MBLPdm — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#3 (comment)](https://github.com/saqi/Maybank/pull/3#issuecomment-427684609)>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABieU8GeBPZZX4EElS3qsmxmINi78hlZks5uimTIgaJpZM4XMA3l> .

Sure, good luck and let me know if you have any issues/ want any ideas.

adampaulukanis commented:
Closes #5

Have a good day.

Commit:
Another attempt

./Images/pic{1,2,3}.png
I had to modify these a little bit.

Changes in index.html and style.css

Commit:
Add / Replace carrousel pictures

adampaulukanis commented:
Hello again.

I have added:
- new photos. Replaced the ones I made for carrousel.
- text slides up now

Checking the site in Chrome/FF developer tools make me cry when I switch to IPad emulation. I do not have it and I do not know why it does not work there. Any Ideas @saqi ? ;>
Hmm. I have just checked it again and it started to work…WTF?

And I have an idea? Currently next/prev buttons are rather small. How about making them bigger?

Commit:
Add alt tag to some images

Checked the page against https://validator.w3.org/ and found out there are some errors to fix ;-)

Now there are 3 left which gonna stay.

Commit:
Slideshow fixed (sort of)

I got rid of prev/next buttons and added loader bar under. Looks cool.

adampaulukanis commented:
it closes #4.

Commit:
Small change to menu

Change appearance.

Commit:
Create slideup.js

saqi commented:
Hello hello,

Carousel section:
- Pics 1 - 5 are fine, 6 isn't ;)
- I think the prev next arrows looked better 
- I also got the same issue when exploring different devices, I think that loader is using too much memory? Anyway I kept the page open for some time and found that after a little while the change of pics and loader go out of sync so the issue occurs on laptop too and I have 16GB RAM :) Probably best to remove it for now
- Delay between each image is quite short, extending it for at least 2 seconds would do it some good :p 

:+1: for text sliding up, I want to start perfecting the small bits so I'll try get the original pics so that you won't need to block out the text. Also I will get appropriate/real text for the sections. If you can make 1 change for this section is to insert those 3 images again(on which the text slides up) to fill the page up just so that the client will be able to see how the webpage will be filled up with those extra images.
Summary:
- I'll try get original pics
- I'll get appropriate text for you to paste in
- Please insert another row of the text slide up images

Once **all** of the above bullet points are done I'll merge with master and we can concentrate on header and menu hopefully, thank you :1st_place_medal: 

Don't worry about prev next buttons working for now, I might want to refactor the way the images change but it's working and that's all that matters for now

saqi commented:
Images:
[Image 1 - Specialist Design](https://www.maybankassociates.com/wp-content/uploads/2016/02/20140603_111810-e1457097984242-650x400.jpg)
[Image 2 - Gilding](https://www.maybankassociates.com/wp-content/uploads/2016/02/1970710_288128421344215_1604675810_n-650x400.jpg)
[Image 3 - Sign writing](https://www.maybankassociates.com/wp-content/uploads/2014/06/P1000214-1024x768.jpg)

Text for these images:
Specialist Design: With years of experience in specialist decorating, we have worked on a variety of projects. Whatever your requirements, you can expect a high quality finish.
Gilding: We specialise in all types of internal and external gilding. Our portfolio boasts work done from Churches and Cottages to Mosques and Mansions. 
Sign Writing: We design, manufacture and install signs, including advertising signs for shops, businesses and public facilities as well as signs for transport systems.

Commit:
WIP #1

saqi commented:
Hello,

Is it possible to have automatic slideshow with prev next buttons? It was only the loader that was messing it up, the changing of images was fine :)

saqi commented:
I've coded up a buggy carousel, if you want you can fix them as the code is quite simple:
https://codepen.io/Saqalain/full/ePGpMb/

Else I've forked and slightly changed someone else's where the code may be a little more complex but works perfect:
https://codepen.io/Saqalain/full/OBxMmw/

The second one is related to this one which explains the code very well, worth checking out if you opt for the second link. The codepen also has a link the original author based it upon.

Hope it helps!

saqi reviewed (changes requested):
- automatic slideshow with buttons
- original pics
- appropriate text for you to paste in
- another row of the text slide up images

Commit:
Changes

adampaulukanis commented:
Hello, Sorry I was lazy.
Could you please have a look now and write what you want to be changed?
Slide show:
- prev/next work (or seems to work to me;))
- automatic slideshow stops after 2nd image. Any idea why? 
I need to reboot my computer now ;)
Take care.

Commit:
SlideShow should work now ;-)

adampaulukanis commented:
I hope now **slideshow** works as it should.
Slides change every 7 seconds, clicking **prev/next** buttons delay displaying slides by 10 seconds and slideshow starts slideshow again.

saqi commented:
Hello,

Well done on the slideshow, just last improvements:
Prev and next stick out and I think just having < & > at a bigger font size or << & >> would look better
I'll try get the first original image so we don't get that double text effect 
Appropriate text for other slides (I think 3 has 3.3333 lol)
The sideshow doesn't work on my phone, I can only scroll through the pics, I'll send you link to see if it's the same for you

Image with text slide up is perfect now, just add another row of the same as per original design as they will want more of them

And that's pretty much it, once done we can merge and move on to the easy stuff ;)

Commit:
Slideshow *prev/next* buttons are changed a bit

The buttons are changed. Now there are bigger and only chevrons last.

Commit:
Changes as suggested

adampaulukanis commented:
Hey! 

- Slideshow's *prev/next* button changed. I am not sure if this is what you really wanted though.
- I do not know what text should go there as we do not have pictures and descriptions. Those are only examples. Nothing which should really stay there.
- Slideshow: on mobile phones is done that way because I thought it was better. Of course I can change it and it will look differently as you wish. Could you design the look of it? Have you got any ideas about that or anything else? ;-)
- SlideUps: there are now 6 boxes, I called them *grid* and made them with *flexbox* to introduce more confusion.
- And last but not least: I would like to remove the bit saying "© 2018 Website designed by Saq and Adam" or hide it in the comments.

saqi commented:
You've got the right idea with the buttons, if they could look like the original in image 1 for example or you can see it https://maybank.netlify.com there as well

True, I'll work on getting some appropriate text there

Can we not get the same functionality in mobile? Or is it my browser that's displaying it like that

Slideups are all good but I think the line would look better in the overflow (beneath the image)

Yh I agree, that came in automatically when I deployed with netlify

Commit:
Change slideshow buttons to look like ><

Also being there I changed the size and shapes of *prev/next* buttons to
look like the previous version.

Commit:
Move down the line in Grid

Previously the line separating the heading from the text was visible.
Now it is hidden and to see it me must hover over the grid element. Does
it make any sens? ;)
